prusti/tests/extern_spec/pass/set.rs

```diff
@@ -0,0 +1,52 @@
+extern crate prusti_contracts;
+use prusti_contracts::*;
+
+use std::collections::HashSet;
+use std::borrow::Borrow;
+use std::hash::Hash;
+use std::cmp::Eq;
+use std::option::Option;
+
+#[extern_spec]
+impl<T> Option<T> {
+    #[pure]
+    #[ensures(matches!(*self, Some(_)) == result)]
+    pub fn is_some(&self) -> bool;
+
+    #[pure]
+    #[ensures(self.is_some() ==> !result)]
+    #[ensures(!self.is_some() ==> result)]
```

Comment:
Maybe this doesn't make a difference in practice, but why not `result ==> !self.is_some()`? I think you wrote it in that order in the `LinkedList` spec.

Reply:
Hmm I'm not sure why I wrote it this way, but no, I don't think that in practice it makes a difference. I think I will just change this to `result != self.is_some()`